go.mod

```diff
@@ -127,6 +127,8 @@ replace (
 
 replace github.com/ajeddeloh/go-json => github.com/coreos/go-json v0.0.0-20220810161552-7cce03887f34
 
+replace github.com/kubermatic/machine-controller => github.com/kubermatic/machine-controller v1.56.1-0.20230601122944-bfd2fd7cf541
```

Comment:
Why do we need this explicit replace directive now?

Reply:
On updating the `kubeone` package and running `go mod tidy` the MC version gets bumped to v1.56.2. Ideally should be > v1.56.1-0.20230601122944-bfd2fd7cf541(latest commit) but it isn't.

Reply:
As much as `replace` is a quick fix for this, do we have any idea why `go mod` is starting to resolve this differently? Afaik the previous KubeOne version also has a dependency on machine-controller, so I think we should try to figure out why this is happening?

Reply:
Sure you can create a separate ticket if you want to investigate. Right now all I want is the latest version of KubeOne to test a few things with external clusters; which this PR should do.

Reply:
Please add at least a comment explaining why we are _seemingly_ downgrading the MC with a replace directive.

> Sure you can create a separate ticket if you want to investigate.

:zipper_mouth_face:

Reply:
> Sure you can create a separate ticket if you want to investigate. Right now all I want is the latest version of KubeOne to test a few things with external clusters; which this PR should do.

I'm sorry but this is not someone else's responsibility to track. You want to put a hack in place, that's okay temporarily, but it's your hack, so please track it accordingly.